### PR TITLE
vendor: github.com/docker/docker 1a1f3cff45ec (master, v27.0-dev)

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/pull/47950
- relates to https://github.com/moby/moby/pull/48023
- relates to https://github.com/moby/moby/pull/48040
- relates to https://github.com/moby/moby/pull/48041
- relates to https://github.com/docker/cli/pull/5145

### vendor: github.com/docker/docker 1a1f3cff45ec (master, v27.0-dev)

Rewrite local code to use the new container.Ulimit alias to start
transitioning away from direct uses of go-units.Ulimit.

full diff: https://github.com/docker/docker/compare/v27.0.0-rc.2...1a1f3cff45ec5aba4a520fae88b4f929eab8b3e8

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@laurazard @vvoland PTAL 🤗

File: cli/command/container/stats_helpers.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (s *stats) isKnownContainer(cid string) (int, bool) {
 	return -1, false
 }
 
-func collect(ctx context.Context, s *Stats, cli client.APIClient, streamStats bool, waitFirst *sync.WaitGroup) {
+func collect(ctx context.Context, s *Stats, cli client.ContainerAPIClient, streamStats bool, waitFirst *sync.WaitGroup) {
```
Comment (thaJeztah) [inline cli/command/container/stats_helpers.go:53]: Changed this to a narrower interface, as it doesn't need all of api-client

File: cli/command/container/stats_unit_test.go [hunk 2]
```diff
@@ -1,30 +0,0 @@
-package container
-
-import (
-	"testing"
-
-	"github.com/docker/docker/api/types"
-)
-
-func TestCalculateBlockIO(t *testing.T) {
```
Comment (thaJeztah) [inline cli/command/container/stats_unit_test.go:9]: Merged this test into the other test-file (`stats_helpers_test.go`)

**Review by sam-thibault — APPROVED** [review]

**Comment by thaJeztah** [comment]
May be doing one more update to include https://github.com/moby/moby/pull/48011

**Comment by thaJeztah** [comment]
Updated, with https://github.com/moby/moby/pull/48011 included

**Review by vvoland — APPROVED** [review]

**Pull request merged**